languages/ramldoc/src/main/kotlin/io/vrap/codegen/languages/ramldoc/extensions/VrapExtensions.kt

```diff
@@ -106,3 +247,54 @@ class ObjectInstanceSerializer : JsonSerializer<ObjectInstance>() {
         gen.writeEndObject()
     }
 }
+
+fun Instance.toYaml(): String {
+    var example = ""
+    val mapper = YAMLMapper()
+    mapper.disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER)
+
+    val module = SimpleModule()
+    module.addSerializer(ObjectInstance::class.java, ObjectInstanceSerializer())
+    module.addSerializer<Instance>(ArrayInstance::class.java, InstanceSerializer())
+    module.addSerializer<Instance>(IntegerInstance::class.java, InstanceSerializer())
+    module.addSerializer<Instance>(BooleanInstance::class.java, InstanceSerializer())
+    module.addSerializer<Instance>(StringInstance::class.java, InstanceSerializer())
+    module.addSerializer<Instance>(NumberInstance::class.java, InstanceSerializer())
+    mapper.registerModule(module)
+
+    if (this is ObjectInstance) {
+        try {
+            example = mapper.writeValueAsString(this)
+        } catch (e: JsonProcessingException) {
+        }
+    } else {
+        example = mapper.writeValueAsString(this.value)
+    }
+
+    return example.trim()
+}
+
+fun Instance.toJson(): String {
+    var example = ""
+    val mapper = ObjectMapper()
+
+    val module = SimpleModule()
+    module.addSerializer(ObjectInstance::class.java, ObjectInstanceSerializer())
```

Comment:
it might come handy to create a module factory here

Reply:
Yeah you may be right. But I was too lazy 😈